Sync tab settings

jeremypw opened the pull request:
Fixes #1340 

* Ensure the settings, appmenu preferences, format bar and tab popover stay in sync and are applied to doc when editorconfig not overriding
* Allow overriding tab width and tab style separately
* Ensure correct settings displayed and used when switching documents while editorconfig active.
* Ensure correct symbol used when drawing spaces when switching between spaces and tabs

There is a pre-existing issue that the range of the tab width spinbutton is up to 24 while the maximum tab width set by editorconfig plugin is 16.  This should be resolved here or later.

As this PR changes a plugin, it needs to be installed for testing.

zeebok commented on src/Widgets/FormatBar.vala:
```diff
@@ -180,27 +184,42 @@ public class Code.FormatBar : Gtk.Box {
+            if (!tab_width_set_by_editor_config) {
+                Scratch.settings.set_int (
+                    "indent-width",
+                    ((int)width_spinbutton.@value).clamp (2, 16)
```
If we are clamping to 16 both in editorconfig and here, should we just remove this clamp and limit the spinbutton?

jeremypw replied:
I don't see why not. Do you agree that `16` is a reasonable maximum for the tab width?

zeebok reviewed (approved):
Code looks good and works as advertised. Besides my one question I think this is good to merge.

jeremypw commented:
@zeebok I have removed the clamp as you suggested and added a corresponding forced range to the schema.  Also changed the range in the plugin to correspond.

jeremypw commented:
I notice that `gedit` only allows tabs in the range 2 - 8 and also only even numbers.

zeebok reviewed (approved):
> I notice that `gedit` only allows tabs in the range 2 - 8 and also only even numbers.

Even seems nice but I don't feel strongly enough to not approve this. I think this is great and good customization.
